shell/platform/android/platform_view_android_jni.cc

```diff
@@ -207,9 +208,16 @@ static void RunBundleAndSnapshot(
 
   if (bundlepath.size() > 0) {
     // If we got a bundle path, attempt to use that as a directory asset
-    // bundle.
-    asset_manager->PushBack(std::make_unique<blink::DirectoryAssetBundle>(
-        fml::OpenFile(bundlepath.c_str(), fml::OpenPermission::kRead, true)));
+    // bundle or a zip asset bundle.
+    const auto file_ext_index = bundlepath.rfind(".");
+    if (bundlepath.substr(file_ext_index) == ".zip") {
```

Comment:
You don't need these checks. You can just attempt to push back the two instances. If they are invalid, they will be rejected in the push call.

Reply:
That is,
```
asset_manager->PushBack(std::make_unique<blink::ZipAssetStore>(...)
asset_manager->PushBack(std::make_unique<blink::DirectoryAssetBundle>(...)
```

should work just fine :)

Reply:
Here, bundlepath could be either a zip file or a directory. So what you're proposing would require making ZipAssetBundle resilient when it gets a directory instead of a zip file, so that control flow can follow through to DirectoryAssetBundle. Picking the right asset store right here seems easier. Let me know what you think...

Reply:
I am saying that it is already resilient :) But your call.

Reply:
FYI, unfortunately, it's not resilient :) It will crash when trying to open a zip file that doesn't exist.